actgen/nnutils.py

```diff
@@ -94,23 +94,38 @@ def soft_copy_from(self, other, tau=0.1):
 
 
 class MLP(Network):
-    def __init__(self, n_inputs, n_outputs, n_hidden_layers, n_units_per_layer):
+    def __init__(self, n_inputs, n_outputs, n_hidden_layers, n_units_per_layer, 
+                use_dropout=False, dropout_rate=0.5):
```

Comment:
I think we want the following API: `dropout=False` is the default, but it can be overridden with `dropout=0.5`, which would automatically enable dropout.